### PR TITLE
Translated Comments from German to English

### PR DESCRIPTION
**Pull request by arjun-kshirsagar**
# Pull Request: Translation of German Comments to English

## Overview

This pull request (PR) addresses the issue of German comments found in various source files within the LibreOffice repository. The primary goal is to translate these comments into English to enhance code readability and maintainability for all contributors.

## Files Translated

The following files have been updated with translated comments:

- **`core/src/main/java/org/libreoffice/lots/db/SchemaDatasource.java`**
- **`core/src/main/java/org/libreoffice/lots/db/Search.java`**
- **`core/src/main/java/org/libreoffice/lots/db/SearchStrategy.java`**
- **`core/src/main/java/org/libreoffice/lots/db/SimpleDataset.java`**
- **`core/src/main/java/org/libreoffice/lots/db/ThingyDatasource.java`**
- **`core/src/main/java/org/libreoffice/lots/db/UnionDatasource.java`**

## Summary of Changes

- Translated German comments to English to improve understanding and accessibility for the international development team.
- Ensured that the meaning and context of the original comments were preserved during translation.

## Why This Change?

Translating comments to English is crucial for:
- Improving code readability for international developers.
- Ensuring consistency in documentation across the codebase.
- Facilitating easier code maintenance and onboarding of new contributors.

## Related Issues

- Resolves issue partially [#393 ]

### CODE REVIEW

**Comment by arjun-kshirsagar** [comment]
I had changed the comments on the GitHub itself, so you can ignore the checks as no single line of code was changed apart from the German comments.

@thorstenb @smehrbrodt

File: core/src/main/java/org/libreoffice/lots/db/Search.java [hunk 1]
```diff
@@ -180,9 +178,9 @@ private static String formatQuery(String query)
   }
 
   /**
-   * Nimmt ein Template für eine Suchanfrage entgegen (das Variablen der Form "${suchanfrageX}"
-   * enthalten kann) und instanziiert es mit Wörtern aus words, wobei nur die ersten wordcount
-   * Einträge von words beachtet werden.
+   * Takes a template for a search query (which may contain variables of the form "${suchanfrageX}")
+   * and instantiates it with words from words, considering only the first wordcount
+   * entries of words.
```
Comment (laenion) [inline core/src/main/java/org/libreoffice/lots/db/Search.java:183]: "suchanfrageX" should also be translated, such as "searchqueryX".

> instantiates it with words from words

That's difficult to parse in English - in German you knew that the English term refers to the variable name, but in pure English that context gets lost. How about "instantiates it with a list of words"?

File: core/src/main/java/org/libreoffice/lots/db/SearchStrategy.java [hunk 2]
```diff
@@ -32,35 +32,32 @@
...
   private Map<Integer, List<Query>> mapWordcountToListOfQuerys;
 
   /**
-   * {@link #mapWordcountToListOfQuerys} wird per Referenz eingebunden und
-   * entsprechende Ergebnisse aus dieser Map werden von {@link #getTemplate(int)}
-   * zurückgeliefert.
+   * {@link #mapWordcountToListOfQuerys} is included by reference and
+   * corresponding results from this map are returned by {@link #getTemplate(int)}.
    */
+
```
Comment (laenion) [inline core/src/main/java/org/libreoffice/lots/db/SearchStrategy.java:49]: There are several additional newlines in this pull request without any obvious pattern when they are added and when they are skipped...

File: core/src/main/java/org/libreoffice/lots/db/UnionDatasource.java [hunk 3]
```diff
@@ -88,12 +85,11 @@ public UnionDatasource(Map<String, Datasource> nameToDatasource,
...
-     * Anmerkung: Die folgende Bedingung ist "unnötig" streng, aber um sie
-     * aufzuweichen (z.B. Gesamtschema ist Vereinigung der Schemata) wäre es
-     * erforderlich, einen Dataset-Wrapper zu implementieren, der dafür sorgt, dass
-     * alle Datasets, die in QueryResults zurück- geliefert werden das selbe Schema
-     * haben. Solange dafür keine Notwendigkeit ersichtlich ist, spare ich mir diesen
-     * Aufwand.
+     * Note: The following condition is "unnecessarily" strict, 
+     * but to relax it (e.g., the overall schema is the union of the schemas) 
+     * it would be necessary to implement a dataset wrapper that ensures that 
+     * all datasets returned in query results have the same schema.
```
Comment (laenion) [inline core/src/main/java/org/libreoffice/lots/db/UnionDatasource.java:91]: > query results

This should be `QueryResults`.

**Review by laenion — COMMENTED** [review]
In general the quality of the translation is quite good (though I assume it is at least based on a machine translation?), so just adding a few minor comments.

**Pull request closed without merging**